daemon/commit.go

```diff
@@ -70,6 +74,21 @@ func (daemon *Daemon) Commit(container *Container, repository, tag, comment, aut
 	}
 	defer rwTar.Close()
 
+	blobStore := daemon.graph.BlobStore()
+	diffWriter, err := blobStore.NewWriter(crypto.SHA256)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get new writer for rootfs diff: %s", err)
+	}
+	defer func() {
+		if err != nil {
```

Comment:
I think you are right in this case since the named return value is considered in the same scope as the previous redeclaration. However if the redeclaration happened in an nested block the behavior would change, one of the more confusing aspects of the go language.